### PR TITLE
Add a volume property to the web audio

### PR DESCRIPTION
**Pull request by skyproudmanp**
Summary: Implement a "volume" property on Web Audio to set/attenuate the volume before mixing with the Media Player & TTS audio output (the volume of the latter will not be affected)
Type: Feature
Test Plan: UT, CT and test on XiOne US 
Jira: RIALTO-595

### CODE REVIEW

**Review by skywojciechowskim — APPROVED** [review]

File: source/RialtoGStreamerWebAudioSink.cpp [hunk 1]
```diff
@@ -262,26 +267,40 @@ static gboolean rialto_web_audio_sink_event(GstPad *pad, GstObject *parent, GstE
...
         {
-            if (!sink->priv->m_webAudioClient->play())
+            result = true;
+
+            if (priv->isVolumeQueued)
             {
-                GST_ERROR_OBJECT(sink, "Failed to play web audio");
+                if (!priv->m_webAudioClient->setVolume(priv->volume))
+                {
+                    GST_ERROR_OBJECT(sink, "Failed to set volume");
```
Comment (skyfroxl) [inline source/RialtoGStreamerWebAudioSink.cpp:282]: No coverage, can we add a test for this

File: source/RialtoGStreamerWebAudioSink.cpp [hunk 2]
```diff
@@ -321,6 +386,23 @@ static void rialto_web_audio_sink_set_property(GObject *object, guint propId, co
...
+        priv->volume = g_value_get_double(value);
+        if (!kClient || !kClient->isOpen())
+        {
+            GST_DEBUG_OBJECT(object, "Enqueue volume setting");
+            priv->isVolumeQueued = true;
+            return;
+        }
+        if (!kClient->setVolume(priv->volume))
+        {
+            GST_ERROR_OBJECT(object, "Failed to set volume");
```
Comment (skyfroxl) [inline source/RialtoGStreamerWebAudioSink.cpp:401]: No coverage, can we add a test for this

**Review by skywojciechowskim — DISMISSED** [review]

File: source/RialtoGStreamerMSEAudioSink.cpp [hunk 3]
```diff
@@ -345,40 +348,47 @@ static void rialto_mse_audio_sink_get_property(GObject *object, guint propId, GV
         return;
     }
 
-    std::shared_ptr<GStreamerMSEMediaPlayerClient> client = basePriv->m_mediaPlayerManager.getMediaPlayerClient();
+    const std::shared_ptr<GStreamerMSEMediaPlayerClient> &kClient = basePriv->m_mediaPlayerManager.getMediaPlayerClient();
```
Comment (skyfroxl) [inline source/RialtoGStreamerMSEAudioSink.cpp:351]: Missed this previously but I think we should revert back to std::shared_ptr<GStreamerMSEMediaPlayerClient> in these places, MediaPlayerClient can get released (releaseMediaPlayerClient) on a separate thread, taking a copy of the shared_ptr ensures we wait till finished with the object before it gets destructed.

File: source/RialtoGStreamerWebAudioSink.cpp [hunk 4]
```diff
@@ -142,6 +144,8 @@ static gboolean rialto_web_audio_sink_send_event(GstElement *element, GstEvent *
 static GstStateChangeReturn rialto_web_audio_sink_change_state(GstElement *element, GstStateChange transition)
 {
     RialtoWebAudioSink *sink = RIALTO_WEB_AUDIO_SINK(element);
+    RialtoWebAudioSinkPrivate *priv = sink->priv;
+    const std::shared_ptr<GStreamerWebAudioPlayerClient> &kClient = priv->m_webAudioClient;
```
Comment (skyfroxl) [inline source/RialtoGStreamerWebAudioSink.cpp:148]: Probably not so important for webaudio, as the lifecycle of the webaudioclient covers init-?finalize

File: source/RialtoGStreamerMSEAudioSink.cpp (hunk 3 above)
Comment (skyproudmanp) [reply source/RialtoGStreamerMSEAudioSink.cpp:351]: Well spotted!

**Review by skyshaha1 — APPROVED** [review]

**Review by skyfroxl — APPROVED** [review]

**Pull request merged**